src/main/java/jenkins/plugins/logstash/configuration/ElasticSearch.java

```diff
@@ -125,33 +129,48 @@ public int getDefaultPort()
       return 9300;
     }
 
-    @Override
-    public FormValidation doCheckHost(@QueryParameter("value") String value)
+    public FormValidation doCheckUri(@QueryParameter("value") String value)
     {
       if (StringUtils.isBlank(value))
       {
         return FormValidation.warning(Messages.PleaseProvideHost());
       }
       try
       {
-        new URL(value);
+        URI uri = new URI(value);
+        validateUri(uri);
       }
-      catch (MalformedURLException e)
+      catch (URISyntaxException | IllegalArgumentException e)
       {
         return FormValidation.error(e.getMessage());
       }
       return FormValidation.ok();
     }
+  }
 
-    public FormValidation doCheckKey(@QueryParameter("value") String value)
-    {
-      if (StringUtils.isBlank(value))
+  /**
+   * Validates that the given uri has a scheme, a port and a path which is not empty or just "/"
+   *
+   * @param uri
+   * @throws IllegalArgumentException when one of scheme, port or path
+   */
+  public static void validateUri(URI uri) throws IllegalArgumentException
```

Comment:
since we ignore those I'd add a check enforcing empty username and password (in authority part)